src/base/SlipStreamSwapRouter.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.18;
+
+import "solady/src/utils/SafeTransferLib.sol";
+import {ERC20Callee} from "../libraries/ERC20Caller.sol";
+import "../libraries/SlipStreamPoolAddress.sol";
+import {PoolAddressPancakeSwapV3} from "@aperture_finance/uni-v3-lib/src/PoolAddressPancakeSwapV3.sol";
+import {TernaryLib} from "@aperture_finance/uni-v3-lib/src/TernaryLib.sol";
+import {OptimalSwap, TickMath, V3PoolCallee} from "../libraries/OptimalSwap.sol";
+import {PCSV3Immutables, UniV3Immutables} from "./Immutables.sol";
+import {Payments} from "./Payments.sol";
+import "./Callback.sol";
+
+/// @title Optimal Swap Router
+/// @author Aperture Finance
+/// @dev This router swaps through an aggregator to get to approximately the optimal ratio to add liquidity in a UniV3-style
+/// pool, then swaps the tokens to the optimal ratio to add liquidity in the same pool.
+abstract contract SlipStreamSwapRouter is Payments, SlipStreamCallback {
+    using SafeTransferLib for address;
+    using TernaryLib for bool;
+    using TickMath for int24;
+
+    /// @dev Literal numbers used in sqrtPriceLimitX96 = zeroForOne ? MIN_SQRT_RATIO + 1 : MAX_SQRT_RATIO - 1
+    /// = (MAX_SQRT_RATIO - 1) ^ ((MIN_SQRT_RATIO + 1 ^ MAX_SQRT_RATIO - 1) * zeroForOne)
+    uint160 internal constant MAX_SQRT_RATIO_LESS_ONE = 1461446703485210103287273052203988822378723970342 - 1;
```

Comment:
for better readability:
```
/// @dev The maximum value that can be returned from #getSqrtRatioAtTick. Equivalent to getSqrtRatioAtTick(MAX_TICK)
uint160 internal constant MAX_SQRT_RATIO = 1461446703485210103287273052203988822378723970342;
/// @dev Literal numbers used in sqrtPriceLimitX96 = zeroForOne ? MIN_SQRT_RATIO + 1 : MAX_SQRT_RATIO - 1
/// = (MAX_SQRT_RATIO - 1) ^ ((MIN_SQRT_RATIO + 1 ^ MAX_SQRT_RATIO - 1) * zeroForOne)
uint160 internal constant MAX_SQRT_RATIO_LESS_ONE = MAX_SQRT_RATIO - 1
```

Reply:
Can't do it this way because of Solidity compiler restriction: "Only direct number constants and references to such constants are supported by inline assembly. solidity(7615)"